theseus/embodied/robot/joint.py

```diff
@@ -217,3 +217,158 @@ def _rotation_impl(self, angle: torch.Tensor) -> torch.Tensor:
         rot[:, 0, 1] = -rot[:, 1, 0]
         rot[:, 1, 1] = rot[:, 0, 0]
         return rot
+
+
+class _PrismaticJointImpl(Joint):
+    def __init__(
+        self,
+        name: str,
+        parent: int = -1,
+        child: int = -1,
+        origin: Optional[torch.Tensor] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        super().__init__(name, parent, child, origin, dtype)
+
+    def dof(self) -> int:
+        return 1
+
+    @abc.abstractmethod
+    def _translation_impl(self, angle: torch.Tensor) -> torch.Tensor:
+        pass
+
+    def relative_pose(self, angle: torch.Tensor) -> torch.Tensor:
+        trans = self._translation_impl(angle)
+        ret = angle.new_empty(angle.shape[0], 3, 4)
+        ret[:, :, :3] = self.origin[:, :, :3]
+        ret[:, :, 3:] = (
+            self.origin[:, :, :3] @ trans.view(-1, 3, 1) + self.origin[:, :, 3:]
+        )
+        return ret
+
+
+class PrismaticJoint(_PrismaticJointImpl):
+    def __init__(
+        self,
+        name: str,
+        prismatic_axis: torch.Tensor,
+        parent: int = -1,
+        child: int = -1,
+        origin: Optional[torch.Tensor] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        if prismatic_axis.ndim == 1:
+            prismatic_axis = prismatic_axis.view(-1, 1)
+
+        if prismatic_axis.ndim != 2 or prismatic_axis.shape != (3, 1):
+            raise ValueError("The prismatic axis must be a 3-D vector.")
```

Comment:
I imagine this logic of "checking if the input is a valid vector of correct length, and unsqueezing it into a vertical vector if it's 1-D" would be very commonly used, not just within the robot model module. Should this be a piece of logic that can be standardized and deduplicated within an `utils` of some sort?

Reply:
If we anticipate this to be used outside robot model, it would be useful to make an utility function. Otherwise, we should at least delegate this repeated logic to the super class.